DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -227,5 +235,98 @@ public void UnlockProgress(int progressId)
+                        SUM(asp1.TutTime) AS TotalTime,
+                        s.AverageSectionMins AS AverageTime,
+                        CASE WHEN (s.PLAssessPath IS NOT NULL AND cu.IsAssessed = 1) THEN 1 ELSE 0 END AS PostLearningAssessment,
+                        COALESCE (MAX(ISNULL(aa.Score, 0)), 0) AS Outcome,
+                        (SELECT COUNT(AssessAttemptID) AS PLAttempts
+                            FROM AssessAttempts AS aa
+                            WHERE (ProgressID = @ProgressID) AND (SectionNumber = s.SectionNumber)) AS Attempts,
+                        MAX(ISNULL(CAST(ct.Status AS INT), 0)) AS Passed
+                    FROM
+                        aspProgress AS asp1
+                        INNER JOIN Progress AS p ON asp1.ProgressID = p.ProgressID
+                        INNER JOIN Sections AS s
+                        INNER JOIN Tutorials AS t ON s.SectionID = t.SectionID
+                        INNER JOIN CustomisationTutorials AS ct ON t.TutorialID = ct.TutorialID ON asp1.TutorialID = t.TutorialID
+                        INNER JOIN Customisations AS cu ON p.CustomisationID = cu.CustomisationID
+                        LEFT OUTER JOIN AssessAttempts AS aa ON p.ProgressID = aa.ProgressID AND s.SectionNumber = aa.SectionNumber
+                    WHERE
+                        (ct.CustomisationID = p.CustomisationID) AND (p.ProgressID = @ProgressID) AND (s.ArchivedDate IS NULL)
+                        AND (ct.Status = 1 OR ct.DiagStatus = 1 OR cu.IsAssessed = 1)
+                    GROUP BY
+                        s.SectionID,
+                        s.ApplicationID,
+                             s.SectionNumber,
+                             s.SectionName,
+                             s.ConsolidationPath,
+                             s.DiagAssessPath,
+                             s.PLAssessPath,
+                             s.AverageSectionMins,
+                             cu.IsAssessed,
+                             p.CandidateID,
+                             p.CustomisationID,
+                             p.PLLocked
+                    ORDER BY s.SectionNumber",
+                new { progressId }
+            );
+        }
+
+        public IEnumerable<DetailedTutorialProgress> GetTutorialProgressDataForSection(int progressId, int sectionId)
+        {
+            return connection.Query<DetailedTutorialProgress>(
+                @"SELECT
+                        t.TutorialName,
+                        ts.Status AS TutorialStatus,
+                        ap.TutTime AS TimeTaken,
+                        CASE WHEN t.OverrideTutorialMins > 0 THEN t.OverrideTutorialMins ELSE t.AverageTutMins END AS AvgTime,
+                        CASE WHEN (ap.DiagAttempts > 0 AND ct.DiagStatus = 1) THEN ap.DiagLast ELSE NULL END AS DiagnosticScore,
+                        t.DiagAssessOutOf AS PossibleScore
+                    FROM
+                        Progress AS p

```

Comment:
Don't think we need to have the Progress table here, as it looks like its only use is checking the progress Id matches, but we can just apply that constrait to aspProgress.ProgressId instead.

Reply:
It's also used in some of the GROUP BYs.